app/controllers/drops_controller.rb

```diff
@@ -69,8 +69,8 @@ def upvote
   private
 
     def set_drop
-      @drop = Drop.find(params[:id].to_i)
-    end
+      page_not_found unless @drop = Drop.find_by(id: params[:id].to_i)
```

Comment:
The `page_not_found` shouldn't be necessary: Rails' application controller rescues from `ActiveRecord::RecordNotFound` exceptions by displaying the 404 error page  and returning the 404 status code _in production_. In dev mode, it will still show the error.

The error is raised when you use `AR::Base.find(id)` or `find_by!(id: id)` with an _id_ that doesn't exist.

```ruby
def set_drop
  @drop = Drop.find(params[:id])
end
```